app/lib/features/chat/widgets/image_message_builder.dart

```diff
@@ -1,101 +1,138 @@
 import 'dart:convert';
-import 'dart:io';
+import 'dart:typed_data';
 
 import 'package:cached_network_image/cached_network_image.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:string_validator/string_validator.dart';
 
-class ImageMessageBuilder extends StatelessWidget {
+class ImageMessageBuilder extends ConsumerStatefulWidget {
   final types.ImageMessage message;
   final int messageWidth;
+  final bool isReplyContent;
 
   const ImageMessageBuilder({
     Key? key,
     required this.message,
     required this.messageWidth,
+    this.isReplyContent = false,
   }) : super(key: key);
 
+  @override
+  ConsumerState<ImageMessageBuilder> createState() =>
+      _ImageMessageBuilderConsumerState();
+}
+
+class _ImageMessageBuilderConsumerState
+    extends ConsumerState<ImageMessageBuilder> {
+  Uint8List? decodedImage;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (widget.message.metadata?.containsKey('base64') ?? false) {
+      if (widget.message.metadata?['base64'].isNotEmpty) {
+        decodedImage = base64Decode(widget.message.metadata?['base64']);
+      }
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
-    if (message.metadata?.containsKey('base64') ?? false) {
-      if (message.metadata?['base64'].isNotEmpty) {
-        return ClipRRect(
-          borderRadius: BorderRadius.circular(15),
-          child: Image.memory(
-            base64Decode(message.metadata?['base64']),
-            errorBuilder: (
-              BuildContext context,
-              Object url,
-              StackTrace? error,
-            ) {
-              return Text('Could not load image due to $error');
-            },
-            frameBuilder: (
-              BuildContext context,
-              Widget child,
-              int? frame,
-              bool wasSynchronouslyLoaded,
-            ) {
-              if (wasSynchronouslyLoaded) {
-                return child;
-              }
-              return AnimatedOpacity(
-                opacity: frame == null ? 0 : 1,
-                duration: const Duration(seconds: 1),
-                curve: Curves.easeOut,
-                child: child,
-              );
-            },
-            cacheWidth: 256,
-            width: messageWidth.toDouble() / 2,
-            fit: BoxFit.cover,
-          ),
-        );
-      } else {
+    if (decodedImage != null && decodedImage!.isNotEmpty) {
+      if (widget.isReplyContent) {
         return ClipRRect(
           borderRadius: BorderRadius.circular(15),
-          child: const SizedBox(
-            height: 60,
-            width: 60,
-            child: CircularProgressIndicator(strokeWidth: 6),
+          child: Container(
+            constraints: const BoxConstraints(maxHeight: 200, maxWidth: 200),
```

Comment:
why 200? it seems suspicious to me that this isn't relative to the outer size of space we have at all...

Reply:
We should constrain the image size while replying as it could get large enough for overflow. I suppose this max allowance could do that. But if you have suggestions, let me know :)

Reply:
I meant to use max-values based on the size available. Like use MediaQuery and stuff. Like 200 is very small on a screen like mine with 2880px for no clear reason (I know that it isn't actual pixels we use in flutter, but you get the point).

Reply:
though arguably the size is just a preview here to inform the user they picked the right thing, so it doesn't really matter... consider this resolved.